### PR TITLE
PS-6055 Add support for Float and Double data

### PR DESCRIPTION
**Pull request by mohitj1988**
With this PR, pqueryv3.0 will have support for Float and Double data types.

### CODE REVIEW

File: src/random_test.cpp [hunk 1]
```diff
@@ -726,22 +758,26 @@ void Table::CreateDefaultColumn() {
       /* loop untill we select some column */
       while (col_type == Column::COLUMN_MAX) {
 
-        /* columns are 4:3:2:1:1 INT:VARCHAR:CHAR:BLOB:BOOL */
```
Comment (rahulmalik87) [inline src/random_test.cpp:771]: please keep the ratio of (float+ double) less than of int. 
maybe (2:1)

File: src/random_test.cpp [hunk 2]
```diff
@@ -258,6 +258,12 @@ int rand_int(int upper, int lower) {
   return dist(rng);
 }
 
+/* return random float number in the range of upper and lower */
+float rand_float(float upper, float lower) {
+  float r = (rand() / (float)RAND_MAX * (upper - 1)) + lower;
```
Comment (rahulmalik87) [inline src/random_test.cpp]: don't use the rand function.  use some normal distributions.

File: src/random_test.cpp [hunk 3]
```diff
@@ -325,6 +339,14 @@ std::string Column::rand_value() {
     static auto rec = 100 * opt_int(INITIAL_RECORDS_IN_TABLE);
     return std::to_string(rand_int(rec));
     break;
+  case (COLUMN_TYPES::FLOAT):
```
Comment (rahulmalik87) [inline src/random_test.cpp:355]: case FLOAT and DOUBLE seems to be doing the same thing. 
so you can either combine them or see if there is a better way to get rand_float
for them.

**Review by rahulmalik87 — COMMENTED** [review]
add column type handing in load_metadata_from_file.

File: src/random_test.cpp (hunk 2 above)
Comment (mohitj1988) [reply src/random_test.cpp]: Fixed

File: src/random_test.cpp (hunk 3 above)
Comment (mohitj1988) [reply src/random_test.cpp:355]: Fixed

File: src/random_test.cpp [hunk 4]
```diff
@@ -260,8 +262,10 @@ int rand_int(int upper, int lower) {
 
 /* return random float number in the range of upper and lower */
 float rand_float(float upper, float lower) {
-  float r = (rand() / (float)RAND_MAX * (upper - 1)) + lower;
-  return r;
+  std::random_device rd;
```
Comment (rahulmalik87) [inline src/random_test.cpp]: we have already created the seed. Directly call uniform_real_distribution

File: src/random_test.cpp [hunk 5]
```diff
@@ -340,13 +344,25 @@ std::string Column::rand_value() {
...
+  }
   case (COLUMN_TYPES::DOUBLE):
-    static auto rec2 = opt_int(INITIAL_RECORDS_IN_TABLE);
-    return std::to_string(rand_float(rec2));
+  {
+    int rec2 = opt_int(INITIAL_RECORDS_IN_TABLE);
+    rec2 /= 100;
+    std::ostringstream out;
+    out << std::fixed;
+    out << std::setprecision(5) << rand_float(rec2);
```
Comment (rahulmalik87) [inline src/random_test.cpp]: with set precision(5). there will be very hard to find records of double in update/delete queries. you need a better division.

File: src/random_test.cpp (hunk 1 above)
Comment (mohitj1988) [reply src/random_test.cpp:771]: Fixed

File: src/random_test.cpp [hunk 6]
```diff
@@ -726,22 +772,26 @@ void Table::CreateDefaultColumn() {
       /* loop untill we select some column */
       while (col_type == Column::COLUMN_MAX) {
 
-        /* columns are 4:3:2:1:1 INT:VARCHAR:CHAR:BLOB:BOOL */
-        auto prob = rand_int(10);
+        /* columns are 6:2:2:2:2:2:1 INT:FLOAT:DOUBLE:VARCHAR:CHAR:BLOB:BOOL */
```
Comment (rahulmalik87) [inline src/random_test.cpp]: varchar probability becomes very less . make it 4.

File: src/random_test.cpp [hunk 7]
```diff
@@ -325,6 +341,26 @@ std::string Column::rand_value() {
     static auto rec = 100 * opt_int(INITIAL_RECORDS_IN_TABLE);
     return std::to_string(rand_int(rec));
     break;
+  case (COLUMN_TYPES::FLOAT):
+  {
+    float rec1 = opt_int(INITIAL_RECORDS_IN_TABLE);
```
Comment (rahulmalik87) [inline src/random_test.cpp]: use something like static auto rec = opt_int(INITIAL_RECORDS_IN_TABLE)/ 100;

File: src/random_test.cpp [hunk 8]
```diff
@@ -258,6 +260,12 @@ int rand_int(int upper, int lower) {
   return dist(rng);
 }
 
+/* return random float number in the range of upper and lower */
+float rand_float(float upper, float lower) {
```
Comment (rahulmalik87) [inline src/random_test.cpp]: you can add one more argument precision. and move logic here.

**Review by rahulmalik87 — CHANGES_REQUESTED** [review]
please

File: src/random_test.cpp (hunk 5 above)
Comment (mohitj1988) [reply src/random_test.cpp]: Fixed

File: src/random_test.cpp (hunk 6 above)
Comment (mohitj1988) [reply src/random_test.cpp]: Fixed

File: src/random_test.cpp (hunk 7 above)
Comment (mohitj1988) [reply src/random_test.cpp]: Fixed

File: src/random_test.cpp [hunk 9]
```diff
@@ -258,6 +260,15 @@ int rand_int(int upper, int lower) {
   return dist(rng);
 }
 
+/* return random float number in the range of upper and lower */
+std::string rand_float(float upper, float lower, int precision) {
+  std::uniform_real_distribution<> dis(lower, upper);
+  std::ostringstream out;
+  out << std::fixed;
+  out << std::setprecision(precision) << (float)(dis(rng)*100.0)/100.0;
```
Comment (rahulmalik87) [inline src/random_test.cpp]: (float)(dis(rng)*100.0)/100.0. 
can it be done with dis(rng)

**Review by rahulmalik87 — CHANGES_REQUESTED** [review]

File: src/random_test.cpp (hunk 9 above)
Comment (mohitj1988) [reply src/random_test.cpp]: Fixed

**Review by rahulmalik87 — APPROVED** [review]

**Pull request merged**